### ISSUE TITLE
Play an Episode Of Stutter

### ISSUE THREAD

**Issue by gbaldridge** [OmerTu/GoogleHomeKodi#105]
Assistant is stuttering when utilizing the Play an Episode of function.

Sounds like:
ok playing an episode of of TVSHOW

This can be remediated in IFTTT by removing "of" from the response:
ok playing an episode TVSHOW

Assistant will then say it correctly with an "of" even though its removed:
ok playing an episode of TVSHOW

Assuming there's an "extra" "of" hanging around somewhere.

**Comment by gbaldridge** [comment]
This also is happens with a single name show "Ballers". I'm also having trouble with Tosh.0 of course and using different phonetics's isn't working. I'm wondering if there are tools to correct these little issues?

Not sure if its here or deeper:

const kodiPlaySpecificEpisode = (request, response, requestParams) => {
    console.log(`Searching Season ${requestParams.seasonNum}, episode ${requestParams.episodeNum} of Show ID ${requestParams.tvshowid}...`);

**Comment by gbaldridge** [comment]
![balers](https://user-images.githubusercontent.com/34950039/34450878-53dc7270-eccb-11e7-986d-075ae747f707.JPG)
![blackink](https://user-images.githubusercontent.com/34950039/34450880-5c1c8dc6-eccb-11e7-8749-5fa92b8140d4.JPG)

**Comment by gbaldridge** [comment]
Looking at the logs I see that Ballers isnt playing due to there being no unwatched episodes. However since it is failing it locks up and prevents me from giving another command to watch a different tv show. I have to manually turn off the applet and restart it.

Is there an ability to watch a random TV show episode regardless of it being watched?

**Comment by gbaldridge** [comment]
YOUR_GLITCH_SERVER_ADDRESS/playtvshow?q={{TextField}}

This is what appears to continue hanging my applets.

**Comment by keydon** [comment]
can you provide logs from glitch/localNode-JS server?
On what branch are you? master or develop?
When was the last time you updated the app?
Which Kodi version are you running?

**Comment by gbaldridge** [comment]
Yeah the stutter was easily "correctable".

The issue is that there doesnt seem to be a timeout component when a command fails and it remains "stuck". I dont know if this is on Webhooks side or GoogleHomeKodi's side.

1. I was on Glitch and then set it up on my Synology thinking privacy/latency/performance. Not linux savvy enough to know how to get that log but more than happy to if you can direct
2. Master branch. Wasnt aware of development I guess
3. Yesterday was the first day ive setup and used the app
4. Kodi 17.6 and 18 both ive tried on multiple boxes

**Comment by keydon** [comment]
i cannot reproduce your behavior. tried it also with different kinds of errors.

so about the logs, how do you start the server? somewhere there needs to be
```
node server.js
```
change it to
```
node server.js > /full/path/to/a/logfile/of/your/choice.txt
```

please post the contents of that file, after reproducting the blocking behavior and waiting another minute extra.

**Comment by kingofslackerz** [comment]
My stutter is a bit different, GHome is adding an extra "of" even though that's in my phrase in IFTTT.  The error shows the text field "of The Office" when "of" is in the phrase and was only stated once in the command.  Quite odd.

**Comment by keydon** [comment]
@kingofslackerz 
my fix in b5db477 should take care of that
